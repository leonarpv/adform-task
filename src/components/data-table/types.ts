import { ReactNode } from 'react'

export type ColumnDefinition<TSource extends {}> = {
    accessor: (source: TSource) => ReactNode
    title: string
}
