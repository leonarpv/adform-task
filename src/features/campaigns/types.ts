export type Campaign = {
    id: number
    name: string
    startDate: string
    endDate: string
    Budget: number
    userId: number
}

export type CampaignUser = {
    id: number
    name: string
}

export type CampaignWithUser = {
    user: CampaignUser | null
} & Campaign

export type CampaignFilters = {
    name?: string
    startDate?: string
    endDate?: string
}
