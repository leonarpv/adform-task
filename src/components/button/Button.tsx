import React, { ButtonHTMLAttributes, DetailedHTMLProps } from 'react'
import classNames from 'classnames'
import './button.scss'

type Props = DetailedHTMLProps<
    ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
> & {
    secondary?: boolean
}

const Button = ({ secondary, ...rest }: Props) => {
    const className = classNames('button', {
        'button--secondary': secondary,
    })

    return <button className={className} {...rest} />
}

export default Button
