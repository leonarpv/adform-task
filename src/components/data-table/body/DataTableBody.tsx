import React from 'react'
import { ColumnDefinition } from '../types'
import DataTableBodyRow from './DataTableBodyRow'

type Props<TSource> = {
    columns: ColumnDefinition<TSource>[]
    emptyText?: string
    source: TSource[]
}

const DataTableBody = <TSource extends {}>({
    columns,
    emptyText,
    source,
}: Props<TSource>) => {
    const emptyContent = (
        <tr className="data-table__row">
            <td className="data-table__empty-content" colSpan={columns.length}>
                {emptyText || 'No Data'}
            </td>
        </tr>
    )

    const bodyContent =
        source.length === 0
            ? emptyContent
            : source.map((s, index) => (
                  <DataTableBodyRow
                      key={index}
                      columns={columns}
                      sourceEntry={s}
                  />
              ))

    return <tbody className="data-table__body">{bodyContent}</tbody>
}

export default DataTableBody
