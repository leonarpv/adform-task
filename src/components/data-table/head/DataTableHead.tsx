import React from 'react'
import DataTableHeadCell from './DataTableHeadCell'
import { ColumnDefinition } from '../types'

type Props<TSource> = {
    columns: ColumnDefinition<TSource>[]
}

const DataTableHead = <TSource extends {}>({ columns }: Props<TSource>) => {
    const cells = columns.map((c, index) => (
        <DataTableHeadCell column={c} key={index} />
    ))

    return (
        <thead className="data-table__head">
            <tr className="data-table__row">{cells}</tr>
        </thead>
    )
}

export default DataTableHead
