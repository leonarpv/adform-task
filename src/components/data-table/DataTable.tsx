import React from 'react'
import DataTableHead from './head/DataTableHead'
import { ColumnDefinition } from './types'
import DataTableBody from './body/DataTableBody'
import './dateTable.scss'

type Props<TSource> = {
    columns: ColumnDefinition<TSource>[]
    emptyText?: string
    source: TSource[]
}

const DataTable = <TSource extends {}>({
    columns,
    emptyText,
    source,
}: Props<TSource>) => {
    return (
        <table className="data-table">
            <DataTableHead columns={columns} />
            <DataTableBody
                columns={columns}
                emptyText={emptyText}
                source={source}
            />
        </table>
    )
}

export default DataTable
