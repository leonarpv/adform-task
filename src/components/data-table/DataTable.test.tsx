import { shallow } from 'enzyme'
import React from 'react'
import Button from 'components/button/Button'
import DataTable from './DataTable'
import DataTableHead from './head/DataTableHead'
import DataTableBody from './body/DataTableBody'
import { ColumnDefinition } from './types'

describe('DataTable', () => {
    it('should render properly', () => {
        const columns: ColumnDefinition<string>[] = []
        const source: string[] = []

        const wrapper = shallow(<DataTable columns={columns} source={source} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should render table', () => {
        const columns: ColumnDefinition<string>[] = []
        const source: string[] = []

        const wrapper = shallow(<DataTable columns={columns} source={source} />)

        expect(wrapper.find('.data-table').exists()).toBe(true)
        expect(wrapper.find(DataTableHead).exists()).toBe(true)
        expect(wrapper.find(DataTableBody).exists()).toBe(true)
    })
})
