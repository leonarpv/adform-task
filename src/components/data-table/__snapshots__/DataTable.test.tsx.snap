// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable should render properly 1`] = `ShallowWrapper {}`;
